src/main/java/com/google/devtools/common/options/OptionsParserImpl.java

```diff
@@ -488,16 +492,16 @@ private OptionsParserImplResult parse(
       Function<OptionDefinition, String> sourceFunction,
       ParsedOptionDescription implicitDependent,
       ParsedOptionDescription expandedFrom,
-      List<String> args,
-      OptionsData fallbackData)
+      List<ArgAndFallbackData> args)
       throws OptionsParsingException {
     List<String> unparsedArgs = new ArrayList<>();
     List<String> unparsedPostDoubleDashArgs = new ArrayList<>();
     List<String> ignoredArgs = new ArrayList<>();
 
-    Iterator<String> argsIterator = argsPreProcessor.preProcess(args).iterator();
+    Iterator<ArgAndFallbackData> argsIterator = argsPreProcessor.preProcess(args).iterator();
     while (argsIterator.hasNext()) {
-      String arg = argsIterator.next();
+      ArgAndFallbackData argAndFallbackData = argsIterator.next();
+      String arg = argAndFallbackData.arg;
```

Comment:
I think it's clearer to proactively "unpack" the fallback data here too since it's referenced in multiple places.